aws/src/main/java/org/apache/iceberg/aws/s3/signer/S3V4RestSignerClient.java

```diff
@@ -81,14 +71,12 @@ public abstract class S3V4RestSignerClient
   private static final String SCOPE = "sign";
 
   @SuppressWarnings("immutables:incompat")
-  private static volatile ScheduledExecutorService tokenRefreshExecutor;
+  // FIXME auth manager is not properly closed
```

Comment:
Not new to this PR; the executor wasn't being closed either. I think ideally this component should share the sane REST client and the same Auth Manager as the catalog, but that's far from trivial to implement.

Reply:
I think we should make this AutoClosable and make sure we close if it is called (even though it won't be right now).

Reply:
Do you mind if we do that in a follow-up PR? I can work on it immediately after this one is merged.